scheduler/replica_scheduler.go

```diff
@@ -146,17 +146,31 @@ func getNodesWithEvictingReplicas(replicas map[string]*longhorn.Replica, nodeInf
 func (rcs *ReplicaScheduler) getDiskCandidates(nodeInfo map[string]*longhorn.Node, nodeDisksMap map[string]map[string]struct{}, replicas map[string]*longhorn.Replica, volume *longhorn.Volume, requireSchedulingCheck bool) (map[string]*Disk, util.MultiError) {
 	multiError := util.NewMultiError()
 
-	nodeSoftAntiAffinity, err :=
-		rcs.ds.GetSettingAsBool(types.SettingNameReplicaSoftAntiAffinity)
+	nodeSoftAntiAffinity, err := rcs.ds.GetSettingAsBool(types.SettingNameReplicaSoftAntiAffinity)
 	if err != nil {
 		logrus.Errorf("error getting replica soft anti-affinity setting: %v", err)
 	}
+	if volume.Spec.ReplicaSoftAntiAffinity != longhorn.ReplicaSoftAntiAffinityDefault &&
+		volume.Spec.ReplicaSoftAntiAffinity != "" {
+		if volume.Spec.ReplicaSoftAntiAffinity == longhorn.ReplicaSoftAntiAffinityEnabled {
+			nodeSoftAntiAffinity = true
+		} else {
+			nodeSoftAntiAffinity = false
+		}
```

Comment:
Can be. Also, apply this change to `zoneSoftAntiAffinity`

```go
nodeSoftAntiAffinity = volume.Spec.ReplicaSoftAntiAffinity == longhorn.ReplicaSoftAntiAffinityEnabled
```

Reply:
Fixed, thanks!